collector/test/CollectorConfigTest.cpp

```diff
@@ -145,4 +151,54 @@ TEST(CollectorConfigTest, TestEnableExternalIpsRuntimeConfig) {
   EXPECT_TRUE(config.EnableExternalIPs());
 }
 
+TEST(CollectorConfigTest, TestConfigMapTrue) {
+  std::string jsonStr = R"({
+           "networkConnectionConfig": {
+             "enableExternalIps": true
+           }
+         })";
+
+  MockCollectorConfig config;
+  config.MockHandleConfigMapString(jsonStr);
+
+  EXPECT_TRUE(config.EnableExternalIPs());
+}
+
+TEST(CollectorConfigTest, TestConfigMapFalse) {
+  std::string jsonStr = R"({
+           "networkConnectionConfig": {
+             "enableExternalIps": false
+           }
+         })";
+
+  MockCollectorConfig config;
+  config.MockHandleConfigMapString(jsonStr);
+
+  EXPECT_FALSE(config.EnableExternalIPs());
+}
+
+TEST(CollectorConfigTest, TestConfigMapEmpty) {
+  std::string jsonStr = R"({
+           "networkConnectionConfig": {
+           }
+         })";
+
+  MockCollectorConfig config;
+  config.MockHandleConfigMapString(jsonStr);
+
+  EXPECT_FALSE(config.EnableExternalIPs());
+}
+
+TEST(CollectorConfigTest, TestConfigMapInvalid) {
+  std::string jsonStr = R"({
+           "networkConnectionConfig": {
+           }
+         )";
+
+  MockCollectorConfig config;
+  config.MockHandleConfigMapString(jsonStr);
+
+  EXPECT_FALSE(config.EnableExternalIPs());
+}
+
```

Comment:
We can probably rewrite all of these as a single test with something like:
```cpp
TEST(CollectorConfigTest, TestConfigurationFile) {
  std::vector<std::pair<std::string, bool>> tests = {
    { R"({
       "networkConnectionConfig": {
         "enableExternalIps": true
       }
     })", true },
    { R"({
           "networkConnectionConfig": {
             "enableExternalIps": false
           }
         })", false },
    // ...
  };
  
  for (const auto& [input, expected] : tests) {
    CollectorConfig config;
    config.HandleConfigMapString(input);
    EXPECT_EQUAL(config.EnableExternalIPs(), expected);
  }
}
```

Reply:
We are also missing a test for an empty string as input.

Reply:
Done